src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -33,27 +34,54 @@ object RuleUtils {
    * @return indexes built for this plan
    */
   def getCandidateIndexes(indexManager: IndexManager, plan: LogicalPlan): Seq[IndexLogEntry] = {
-    // Map of a signature provider to a signature generated for the given plan.
-    val signatureMap = mutable.Map[String, String]()
-
-    def signatureValid(entry: IndexLogEntry): Boolean = {
-      val sourcePlanSignatures = entry.source.plan.properties.fingerprint.properties.signatures
-      assert(sourcePlanSignatures.length == 1)
-      val sourcePlanSignature = sourcePlanSignatures.head
-
-      if (!signatureMap.contains(sourcePlanSignature.provider)) {
-        val signature = LogicalPlanSignatureProvider
-          .create(sourcePlanSignature.provider)
-          .signature(plan)
-        signatureMap.put(sourcePlanSignature.provider, signature)
-      }
-      signatureMap(sourcePlanSignature.provider).equals(sourcePlanSignature.value)
-    }
+    getLogicalRelation(plan) match {
+      case Some(r) =>
+        // Map of a signature provider to a signature generated for the given plan.
+        val signatureMap: mutable.Map[String, String] = mutable.Map()
```

Comment:
why has this changed from `val signatureMap = mutable.Map[String, String]()`?

Reply:
Fixed it. It had happened due to resolving merge conflicts and an older code had replaced the newer one.